hugr-core/src/extension/type_def.rs

```diff
@@ -13,16 +13,18 @@ use crate::types::TypeBound;
 
 /// The type bound of a [`TypeDef`]
 #[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+#[serde(tag = "b")]
+#[allow(missing_docs)]
 pub enum TypeDefBound {
     /// Defined by an explicit bound.
-    Explicit(TypeBound),
+    Explicit { bound: TypeBound },
```

Comment:
Presumably we will be able to stop doing these annoying renames-to-support serde when we have `hugr-model` :)